pkg/codegen/schema/schema.go

```diff
@@ -128,6 +128,34 @@ func (t *ArrayType) String() string {
 
 func (*ArrayType) isType() {}
 
+// EnumType represents an enum.
+type EnumType struct {
+	// ElementType is the element type of the enum.
+	ElementType Type
+	// Elements are the predefined enum values.
+	Elements []*Enum
+	// ModelAsString indicates whether the enum should be strictly enforced.
+	ModelAsString bool
```

Comment:
`ModelAsString` locks us in with string enums and perhaps isn't totally clear. Maybe something like `Enforced` or `Strict` or `Validated` makes more sense here?